core/src/test/scala/integration/kafka/controller/OfflinePartitionsFromDeletedTopicTest.scala

```diff
@@ -0,0 +1,123 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package integration.kafka.controller
+
+import com.yammer.metrics.core.Gauge
+import kafka.api.IntegrationTestHarness
+import kafka.server.IntegrationTestUtils
+import kafka.utils.TestUtils
+import kafka.zk.KafkaZkClient
+import org.apache.kafka.clients.admin.Admin
+import org.apache.kafka.clients.producer.{KafkaProducer, ProducerRecord}
+import org.apache.kafka.common.serialization.StringSerializer
+import org.apache.kafka.server.metrics.KafkaYammerMetrics
+import org.junit.jupiter.api.Assertions.{assertEquals, assertNotEquals}
+import org.junit.jupiter.api.{BeforeEach, Test, TestInfo}
+import org.slf4j.{Logger, LoggerFactory}
+
+import java.time.Duration
+import java.util.concurrent.TimeUnit
+import scala.jdk.CollectionConverters._
+
+
+class OfflinePartitionsFromDeletedTopicTest extends IntegrationTestHarness {
```

Comment:
There's a new preferred approach to integration testing. See #16500. cc @chia7712

Reply:
I'm happy to use the new harness but I think we'd need a way to disable auto consumer-offset topic creation in it somehow.

Reply:
> I'm happy to use the new harness but I think we'd need a way to disable auto consumer-offset topic creation in it somehow.

I believe if there is no active consumers you shouldn't have offset topic.

Reply:
> I believe if there is no active consumers you shouldn't have offset topic.

I thought so too but the IntegrationTestHarness auto creates the consumer-offset topic by default. Refer https://github.com/apache/kafka/blob/96816a3ed74857cc7631174b2c189ed8b1d9807c/core/src/test/scala/integration/kafka/api/IntegrationTestHarness.scala#L121

I'll try to follow Igor's suggestion and see if we can test this within `TopicDeletionManagerTest` instead to avoid the complexity.

Reply:
> I believe if there is no active consumers you shouldn't have offset topic.

yep, the new test infra (`ClusterTestExtensions`) does NOT create the `__consumer_offsets` in setup phase. By contrast, the zk infra does, and that is why there is a flag `createOffsetsTopic`.

https://github.com/apache/kafka/blob/96816a3ed74857cc7631174b2c189ed8b1d9807c/core/src/test/scala/integration/kafka/api/IntegrationTestHarness.scala#L159

Reply:
> yep, the new test infra (ClusterTestExtensions) does NOT create the __consumer_offsets in setup phase

@chia7712 I suppose that only applies to non `Type.ZK` tests? I reckon `ClusterTestExtensions` for `Type.ZK` uses `ZkClusterInvocationContext`? Wouldn't that use `ClusterConfigurableIntegrationHarness` which extends `IntegrationTestHarness` and auto create offsets topic as a result?

https://github.com/apache/kafka/blob/c2f53a97608683405897c3dcb3bd25a03ceb7dba/core/src/test/java/kafka/test/junit/ZkClusterInvocationContext.java#L275

Reply:
> I suppose that only applies to non Type.ZK tests?

yes. Maybe we should disable that in order to make consistent behavior in new test infra. I will file another jira to address that (and check the possible side effect). I feel this PR can keep using `IntegrationTestHarness`, and then we can do a bit refactor as follow-up

Reply:
https://issues.apache.org/jira/browse/KAFKA-17284